toplevel/role/role.go

```diff
@@ -221,6 +228,61 @@ func (c config) Apply(address string, entriesBytes []byte, dryRun bool, threadPo
 	return nil
 }
 
+// Extracts names of policies referenced within applicable properties of desired roles
+// and updates those properties to only include the names of policies.
+// This is necessary to support policy file references within schemas.
+// The graphql query will return nested object for policy properties but vault api
+// expects only names of policies to be included
+func formatPolicyRefs(desiredRoles []entry) error {
+	// defines applicable properties(s) for each role type to reformat
+	// see https://github.com/app-sre/qontract-schemas/blob/main/schemas/vault-config/role-1.yml
+	typeToProperties := map[string][]string{
+		"approle": {
+			"token_policies",
+			"policies",
+		},
+		"oidc": {
+			"token_policies",
+		},
+		"kubernetes": {
+			"token_policies",
+		},
+	}
+	extracted := []string{}
+	for _, role := range desiredRoles {
+		for _, property := range typeToProperties[role.Type] {
+			policies, ok := role.Options[property].([]interface{})
+			if !ok {
+				return fmt.Errorf("[Vault Role] Failed to convert policy list `%s` within `%s`",
+					property,
+					role.Name,
+				)
+			}
+			for _, policy := range policies {
+				policyMap, ok := policy.(map[interface{}]interface{})
+				if !ok {
+					return fmt.Errorf("[Vault Role] Failed to convert policy map within `%s` attribute of `%s`",
+						property,
+						role.Name,
+					)
+				}
+				policyName, ok := policyMap["name"].(string)
```

Comment:
The operation here does not align with the error message below.

Why? Accessing the map content would happen before the type coercion, and what you are checking here is whether the typo coercion was successful - which is also the right thing to do, given the `interface{}` use. :smile: 

Also, note that `interface{}` can be replaced with its type alias `any` for convenience (some linters also prefer it over the old-style notation as the intent is more explicit).